src/core/mac/mac_frame.hpp

```diff
@@ -1148,70 +1257,73 @@ class Frame : public otRadioFrame
      */
     const uint8_t *GetFooter(void) const;
 
-#if OPENTHREAD_CONFIG_ENABLE_TIME_SYNC
     /**
-     * This method sets the Time IE offset.
+     * This method returns the timestamp when the frame was received.
      *
-     * @param[in]  aOffset  The Time IE offset, 0 means no Time IE.
+     * @returns The timestamp when the frame was received, in microseconds.
      *
      */
-    void SetTimeIeOffset(uint8_t aOffset) { mIeInfo->mTimeIeOffset = aOffset; }
+    const uint64_t &GetTimestamp(void) const { return mInfo.mRxInfo.mTimestamp; }
 
+#if OPENTHREAD_CONFIG_ENABLE_TIME_SYNC
     /**
-     * This method sets the offset to network time.
+     * This method sets the Time IE offset.
      *
-     * @param[in]  aNetworkTimeOffset  The offset to network time.
+     * @param[in]  aOffset  The Time IE offset, 0 means no Time IE.
      *
      */
-    void SetNetworkTimeOffset(int64_t aNetworkTimeOffset) { mIeInfo->mNetworkTimeOffset = aNetworkTimeOffset; }
+    void SetTimeIeOffset(uint8_t aOffset) { mInfo.mTxInfo.mIeInfo->mTimeIeOffset = aOffset; }
 
     /**
-     * This method gets the offset to network time.
+     * This method sets the offset to network time.
      *
-     * @returns  The offset to network time.
+     * @param[in]  aNetworkTimeOffset  The offset to network time.
      *
      */
-    int64_t GetNetworkTimeOffset(void) const { return mIeInfo->mNetworkTimeOffset; }
+    void SetNetworkTimeOffset(int64_t aNetworkTimeOffset)
+    {
+        mInfo.mTxInfo.mIeInfo->mNetworkTimeOffset = aNetworkTimeOffset;
+    }
 
     /**
-     * This method sets the time sync sequence.
+     * This method returns a pointer to the vendor specific Time IE.
      *
-     * @param[in]  aTimeSyncSeq  The time sync sequence.
+     * @returns A pointer to the Time IE, NULL if not found.
      *
      */
-    void SetTimeSyncSeq(uint8_t aTimeSyncSeq) { mIeInfo->mTimeSyncSeq = aTimeSyncSeq; }
+    TimeIe *GetTimeIe(void) { return const_cast<TimeIe *>(const_cast<const Frame *>(this)->GetTimeIe()); }
 
     /**
-     * This method gets the time sync sequence.
+     * This method returns a pointer to the vendor specific Time IE.
      *
-     * @returns  The time sync sequence.
+     * @returns A pointer to the Time IE, NULL if not found.
      *
      */
-    uint8_t GetTimeSyncSeq(void) const { return mIeInfo->mTimeSyncSeq; }
+    const TimeIe *GetTimeIe(void) const;
 
     /**
-     * This method returns the timestamp when the SFD was received.
+     * This method gets the offset to network time.
      *
-     * @returns The timestamp when the SFD was received, in microseconds.
+     * @returns  The offset to network time.
      *
      */
-    uint64_t GetTimestamp(void) const { return mIeInfo->mTimestamp; }
+    int64_t ReadNetworkTimeOffset(void) const { return static_cast<int64_t>(GetTimeIe()->GetTime() - GetTimestamp()); }
```

Comment:
How about `GetNetworkTimeoffset()`?

Reply:
`ReadNetworkTimeoffset()` is called for rx frame, not allowed for tx frame.
`SetNetworkTimeoffset()` is called for tx frame, not allowed for rx frame.

Shall we avoid naming them as a `Get/Set` pair so that it will be obvious that they are implemented very differently?

Reply:
How about `ComputeNetworkTimeOffset()`? It seems to do a bit more than just "read" here.

Reply:
done